### PR TITLE
Initial documentation

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**
- Added initial variant of the LimePay API documentation
- Added Getting-started document

### CODE REVIEW

File: Getting-Started.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+___
+
+### 5. Create Shopper
+
+Whenever you are charging a new user, you must register him as a shopper in LimePay. Once registered you do not need to register him again for every charge after that. If the user is already registered/created you do not need to register him again.
+**Important** to notice is that whenever you are creating a shopper you provide the `walletAddress`.
+This address will be then funded whenever you are creating Payments for that specific shopper.
+**The shopper's wallet address should be the same as the one who signs transactions. If this is not the case, the payment will `fail`.**  
+
+For more information on how to create a shopper, see the `Shoppers` resource in the `API Documentation`
```
Comment (bakasura980) [inline Getting-Started.md]: Consider make `Shoppers` and `API Documentation` links

Comment (bakasura980) [inline Getting-Started.md]: Consider making `Shoppers` and `API Documentation` links

Comment (bakasura980) [inline Getting-Started.md]: Consider making `Shoppers` and `API Documentation` links

File: Getting-Started.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+### 6. Create Payment
+
+Once you have completed the steps above (have Organization, User, API cretendtials, Vendor and a Shopper), you could proceed with creating a payment.
```
Comment (bakasura980) [inline Getting-Started.md]: `Once you have completed the steps above`- > `Once you have been oboarded`

Comment (bakasura980) [inline Getting-Started.md]: `Once you have completed the steps above`- > `Once you have been onboarded`

File: API-Documentation.md [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+The authorization is made by adding `Basic Authorization` header and setting `apiKey` and `secret` as username and password. 
+
+#### 2.1 Creating Vendor
+##### ***Notice**: The Creation of Vendors is **NOT** be accessible for now. Once we make it accessible, we will update the documentation.* 
+In order to create Vendor one should execute the following request:
+
+    POST /v1/vendors
+
+Body data:
+
```
Comment (bakasura980) [inline API-Documentation.md]: "etasgf"

File: API-Documentation.md [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+  
+## Resources
+___
+### 1. API Users
+
+The source is located at `/v1/apiusers` and it is restricted resource. One could execute requests only with his email/password credentials.
+
+The authorization is made by adding `Basic Authorization` header and setting `email` and `password` as username and password. 
+
+#### 1.1 Creating API User
```
Comment (bakasura980) [inline API-Documentation.md]: асдфасдфасдф

File: API-Documentation.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
+API Documentation
```
Comment (vsavovlime) [inline API-Documentation.md:1]: Wow

File: API-Documentation.md [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+============
```
Comment (vsavovlime) [inline API-Documentation.md:2]: Wow 2

File: API-Documentation.md (hunk 5 above)
Comment (vsavovlime) [inline API-Documentation.md:1]: sad

Comment (vsavovlime) [inline API-Documentation.md:1]: dsad

Comment (vsavovlime) [inline API-Documentation.md:1]: dsada

**Review by vsavovlime — COMMENTED** [review]
das

**Review by vsavovlime — COMMENTED** [review]
das

**Review by vsavovlime — COMMENTED** [review]
das

File: Getting-Started.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+Whenever you create a Payment, you must provide the following data: 
+ - `shopper` - The Shopper who will be charged
+ - `items` - Array of `item` objects, containing `description`, `amount` and `quantity`
```
Comment (bakasura980) [inline Getting-Started.md]: lineAmount

File: Getting-Started.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+}
+```
+
+`signedTransactions` is an array of **signed** transactions. More details of how one could sign the transactions can be found in [signing transactions](#signing-transactions) section.
+
+**Important**: The transactions are executed sequentially, meaning that the execution starts with the first transaction in the array and finishes with the last transaction in the array!
+
+### 8. Signing Transactions
+
+This section describes how we can define/implement our `signingTxCallback` function that we need to provide into the `limePayConfig`.
```
Comment (bakasura980) [inline Getting-Started.md]: We dont support it anymore

File: Getting-Started.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+let result = await LimePayWeb.TransactionsBuilder.buildSignedTransactions(result.jsonWallet, password, transactions);
+// The result will be ["rawSignedTransaction0", "rawSignedTransaction1"]
+```
+
+In this example, the first transaction will be `approve` transaction that is going to call the Token contract and `approve` the service contract to `charge` 1 token.
+
+The second transaction will call the service contract and execute the `buySomeService` function.
+
+#### For more information and examples of how to integrate with LimePay, you can check the `sample-projects` repository
```
Comment (bakasura980) [inline Getting-Started.md]: `sample-projects` repository -> link it

File: Getting-Started.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+
+### 6. Create Payment
+
+Once you have completed the steps above (have Organization, User, API cretendtials, Vendor and a Shopper), you could proceed with creating a payment.
+Whenever you create a Payment, you must provide the following data: 
+ - `shopper` - The Shopper who will be charged
+ - `items` - Array of `item` objects, containing `description`, `amount` and `quantity`
+ - `fundTXData` - This object represents the `token` and `ethers` amount that will be needed for the Payment to be executed.
+ - `genericTransactions` - This is an array of `genericTransaction` objects that represent a single generic transaction that the shopper will later on have to sign. The data in the `genericTransaction` consists of `gasPrice`, `gasLimit`, `to`, `functionName` and `params`. These parameters are used for validation of the signed transactions that will be sent from the UI (signed by the shopper).
+
```
Comment (bakasura980) [inline Getting-Started.md:52]: currency parameter also should be included

File: Getting-Started.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+**As a result of the request you will receive `x-lime-token` as a header parameter. This token is used to initialize the check-out form!**
+
+**Summary:** The values that you provide for `tokenAmount` and `etherAmount` are crucial. In the `etherAmount` value you not only set the `ethers` that your signed transactions will charge the shopper, but the `required gas` for the execution of the _signed transactions_[[1](#1)] as-well. 
+
+For more information on how to create a payment, see the `Payments` Resource in the `API Documentation`
+
+[[1](#1)] - The set of transactions that must be executed by the shopper in order for him to buy the service/product. For example, if your dApp charges `tokens`, the first transaction must be `approve` transaction executed at the `Token` contract  and the second one - transaction for buying your product/service. 
+**Important:** The transactions are provided as an array and are executed sequentially in their order in the array!
+
+### 7. Initialize LimePay's Checkout form
```
Comment (bakasura980) [inline Getting-Started.md:65]: You have skipped `2.` step; Restructure the steps

File: API-Documentation.md [hunk 12]
```diff
@@ -0,0 +1,415 @@
...
+
+In order to create API User one should execute the following request:
+
+    POST /v1/apiusers
+
+No body data should be passed.
+
+Returns object of type [API User](#api-user) 
+
+#### 1.2 Getting API User
```
Comment (bakasura980) [inline API-Documentation.md:43]: For certain organization

File: API-Documentation.md [hunk 13]
```diff
@@ -0,0 +1,415 @@
...
+
+In order to get API User one should execute the following request:
+
+    GET /v1/apiusers/:id
+
+Returns object of type  [API User](#api-user)
+
+#### 1.3 Getting all API Users
```
Comment (bakasura980) [inline API-Documentation.md:51]: For certain organization

File: API-Documentation.md [hunk 14]
```diff
@@ -0,0 +1,415 @@
...
+
+In order to get all API Users one should execute the following request:
+
+    GET /v1/apiusers
+
+Returns array of [API User](#api-user) objects.
+
+#### 1.4 Delete API User
```
Comment (bakasura980) [inline API-Documentation.md:59]: For certain organization

File: API-Documentation.md [hunk 15]
```diff
@@ -0,0 +1,415 @@
...
+___
+
+### 2. Vendors
+
+The source is located at `/v1/vendors` and it is restricted resource. One could execute requests only with his `API credentials`.
+
+The authorization is made by adding `Basic Authorization` header and setting `apiKey` and `secret` as username and password. 
+
+#### 2.1 Creating Vendor
+##### ***Notice**: The Creation of Vendors is **NOT** be accessible for now. Once we make it accessible, we will update the documentation.*
```
Comment (bakasura980) [inline API-Documentation.md]: is **NOT** be accessible for now -> is **NOT** accessible for now

File: API-Documentation.md [hunk 16]
```diff
@@ -0,0 +1,415 @@
...
+| `payoutInfo`             | `array`  | Array of [Payout Info](#payout-info) objects                                       | yes      |
+| `vendorPrincipal`        | `object` | [Vendor Principal](#vendor-principal) object                                       | yes      |
+| `emailSetup`             | `object` | [Email Setup](#email-setup) object                                                 | yes      |
+| `defaultPayoutCurrency`  | `string` | Currency code (ISO 4217) for the default payout currency                           | yes      |
+| `vendorStatus`           | `string` | Vendor status. Can be [X,Y,Z] TODO                                                 | yes      |
+| `state`                  | `string` | Vendor state                                                                       | no, if country is not `US` or `CA` |
+
+Returns object of type [Vendor](#vendor) 
+
+#### 2.2 Getting Vendor
```
Comment (bakasura980) [inline API-Documentation.md:102]: API USER credentials

File: API-Documentation.md [hunk 17]
```diff
@@ -0,0 +1,415 @@
...
+
+In order to get Vendor one should execute the following request:
+
+    GET /v1/vendors/:id
+    
+Returns object of type [Vendor](#vendor) 
+
+#### 2.3 Getting All Vendors
```
Comment (bakasura980) [inline API-Documentation.md:110]: API USER credentials

File: API-Documentation.md [hunk 18]
```diff
@@ -0,0 +1,415 @@
...
+
+In order to get all Vendors one should execute the following request:
+
+    GET /v1/vendors
+
+Returns array of [Vendor](#vendor) objects
+
+#### 2.4 Update Vendor
+##### ***Notice**: The Update of Vendor is **NOT** be accessible for now. Once we make it accessible, we will update the documentation.*
```
Comment (bakasura980) [inline API-Documentation.md]: is **NOT** be accessible for now -> is **NOT** accessible for now

File: API-Documentation.md [hunk 19]
```diff
@@ -0,0 +1,415 @@
...
+| `email`                  | `string` | Vendor email                                                                       | yes      |
+| `vatId`                  | `string` | The company’s VAT number                                                           | no, can be undefined if `taxId` is set |
+| `taxId`                  | `string` | The company’s business ID number                                                   | no, can be undefined if `vatId` is set |
+| `payoutInfo`             | `array`  | Array of [Payout Info](#payout-info) objects                                       | yes      |
+| `vendorPrincipal`        | `object` | [Vendor Principal](#vendor-principal) object                                       | yes      |
+| `emailSetup`             | `object` | [Email Setup](#email-setup) object                                                 | yes      |
+| `defaultPayoutCurrency`  | `string` | Currency code (ISO 4217) for the default payout currency                           | yes      |
+| `vendorStatus`           | `string` | Vendor status. Can be [X,Y,Z] TODO                                                 | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: Remove it

File: API-Documentation.md [hunk 20]
```diff
@@ -0,0 +1,415 @@
...
+
+Body data:
+
+| Field                    | Type     | Description                                                                        | Required |
+| ------------------------ | -------- | ---------------------------------------------------------------------------------- | -------- |
+| `firstName`              | `string` | First name of shopper                                                              | yes      |
+| `lastName`               | `string` | Last name of shopper                                                               | yes      |
+| `email`                  | `string` | Shopper email                                                                      | yes      |
+| `vendor`                 | `string` | The ID of the Vendor.                                                              | yes      |
+| `zip`                    | `string` | Shopper's zip code                                                                 | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: remove it

File: API-Documentation.md [hunk 21]
```diff
@@ -0,0 +1,415 @@
...
+| `walletAddress`          | `string` | Shopper's wallet address                                                           | yes      |
+| `vatId`                  | `string` | The company’s VAT number                                                           | no       |
```
Comment (bakasura980) [inline API-Documentation.md]: remove it

File: API-Documentation.md [hunk 22]
```diff
@@ -0,0 +1,415 @@
...
+| `taxId`                  | `string` | The company’s business ID number                                                   | no       |
```
Comment (bakasura980) [inline API-Documentation.md]: remove it

File: API-Documentation.md [hunk 23]
```diff
@@ -0,0 +1,415 @@
...
+
+*Note*: You do not need to specify `vatId` if you specified `taxId` and vice versa.
```
Comment (bakasura980) [inline API-Documentation.md]: remove the whole line

File: API-Documentation.md [hunk 24]
```diff
@@ -0,0 +1,415 @@
...
+    POST /v1/shoppers/reset
+
+Body data:
+
+| Field                | Type     |Description                                                                      | Required   |
+| -------------------- | -------- | ------------------------------------------------------------------------------- | ---------- |
+| `shopperId`          | `string` | The ID of the shopper for which it will reset the malicious attempts            | yes        |
+
+Returns object of type [Shopper](#shopper) 
+___
```
Comment (bakasura980) [inline API-Documentation.md:198]: Add delete/update queries

File: API-Documentation.md [hunk 25]
```diff
@@ -0,0 +1,415 @@
...
+
+    GET /v1/shoppers
+
+Returns array of [Shopper](#shopper) objects 
+
+#### 3.4 Reseting Shopper's malicious attempts
+
+In order to reset the malicious attempts of a Shopper one should execute the following request:
+
+    POST /v1/shoppers/reset
```
Comment (bakasura980) [inline API-Documentation.md:189]: /v1/shoppers/reset -> /v1/shoppers/:id/reset (It should be implemented in that way)

File: API-Documentation.md [hunk 26]
```diff
@@ -0,0 +1,415 @@
...
+| `_id`                    | `string` | The ID of Vendor                                                                   | no       |
+| `name`                   | `string` | Vendor name. (Company's name)                                                      | no       |
+| `email`                  | `string` | Vendor email                                                                       | no       |
+| `country`                | `string` | Country code                                                                       | no       |
+| `city`                   | `string` | Vendor's city                                                                      | yes       |
+| `firstName`              | `string` | First name of Vendor (if individual vendor, not business)                          | yes      |
+| `lastName`               | `string` | First name of Vendor (if individual vendor, not business)                          | yes      |
+| `address`                | `string` | Vendor's street address                                                            | yes       |
```
Comment (bakasura980) [inline API-Documentation.md]: nullable - no

File: API-Documentation.md [hunk 27]
```diff
@@ -0,0 +1,415 @@
...
+
+### Vendor 
+
+| Attribute                | Type     | Description                                                                        | Nullable |
+| ------------------------ | -------- | ---------------------------------------------------------------------------------- | -------- |
+| `_id`                    | `string` | The ID of Vendor                                                                   | no       |
+| `name`                   | `string` | Vendor name. (Company's name)                                                      | no       |
+| `email`                  | `string` | Vendor email                                                                       | no       |
+| `country`                | `string` | Country code                                                                       | no       |
+| `city`                   | `string` | Vendor's city                                                                      | yes       |
```
Comment (bakasura980) [inline API-Documentation.md]: nullable - no

File: API-Documentation.md [hunk 28]
```diff
@@ -0,0 +1,415 @@
...
+| `firstName`              | `string` | First name of Vendor (if individual vendor, not business)                          | yes      |
+| `lastName`               | `string` | First name of Vendor (if individual vendor, not business)                          | yes      |
+| `address`                | `string` | Vendor's street address                                                            | yes       |
+| `phone`                  | `string` | Vendor's phone                                                                     | yes       |
+| `zip`                    | `string` | The zip of the vendor                                                              | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: nullable - no

File: API-Documentation.md [hunk 29]
```diff
@@ -0,0 +1,415 @@
...
+| `state`                  | `string` | Vendor state                                                                       | yes, if country is not `US` or `CA` |
+| `vatId`                  | `string` | The company’s VAT number                                                           | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: nullable - no if taxId is yes

File: API-Documentation.md [hunk 30]
```diff
@@ -0,0 +1,415 @@
...
+| `taxId`                  | `string` | The company’s business ID number                                                   | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: nullable - no if vatId is yes

File: API-Documentation.md [hunk 31]
```diff
@@ -0,0 +1,415 @@
...
+
+
+### Email setup
+
+| Attribute                | Type     | Description                                                                        | Nullable |
+| ------------------------ | -------- | ---------------------------------------------------------------------------------- | -------- |
+| `receipt`                | `object` | Object of type [Receipt](#receipt). Configuration of the receipt email.            | no       |
+| `invoice`                | `object` | Object of type [Invoice](#invoice). Configuration of the invoice email.            | no       |
+
+#### Receipt
```
Comment (bakasura980) [inline API-Documentation.md:339]: Change descriptions

File: API-Documentation.md [hunk 32]
```diff
@@ -0,0 +1,415 @@
...
+| Attribute                | Type     | Description                                                                        | Nullable |
+| ------------------------ | -------- | ---------------------------------------------------------------------------------- | -------- |
+| `subject`                | `string` | The subject that will be shown when sending receipt to shoppers.                   | yes      |
+| `bodyHeaderText`         | `string` | The subject that will be shown when sending receipt to shoppers.                   | yes      |
+
+#### Invoice
```
Comment (bakasura980) [inline API-Documentation.md:345]: Change descriptions

File: API-Documentation.md [hunk 33]
```diff
@@ -0,0 +1,415 @@
...
+| `bodyHeaderText`         | `string` | The subject that will be shown when sending invoice to shoppers.                   | yes      |
+
+### Shopper
+| Attribute                | Type     | Description                                                                        | Nullable |
+| ------------------------ | -------- | ---------------------------------------------------------------------------------- | -------- |
+| `_id`                    | `string` | The ID of the Shopper                                                              | no       |
+| `vendor`                 | `string` | The vendor's name for which the shopper is created.                                | no       |
+| `firstName`              | `string` | Shopper's first name                                                               | no       |
+| `lastName`               | `string` | Shopper's last name                                                                | no       |
+| `zip`                    | `string` | Shopper's zip                                                                      | no       |
```
Comment (bakasura980) [inline API-Documentation.md]: Remove it

File: API-Documentation.md [hunk 34]
```diff
@@ -0,0 +1,415 @@
...
+| `email`                  | `string` | Shopper email                                                                      | no       |
+| `walletAddress`          | `string` | Shopper wallet address                                                             | no       |
+| `vatId`                  | `string` | The shopper’s VAT number                                                           | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: Remove it

File: API-Documentation.md [hunk 35]
```diff
@@ -0,0 +1,415 @@
...
+| `taxId`                  | `string` | The shopper’s TAX ID number                                                        | yes      |
```
Comment (bakasura980) [inline API-Documentation.md]: Remove it

File: API-Documentation.md [hunk 36]
```diff
@@ -0,0 +1,415 @@
...
+| `status`             | `string` | The status of the Payment                                                       | no       |
+| `date`               | `date`   | The date time when the payment was created                                      | no       |
+| `currency`           | `string` | Currency code (ISO 4217) of the amount to be charged                            | no       |
+| `shopper`            | `string` | The shopperID of the shopper that will be charged                               | no       |
+| `vendor`             | `string` | The vendorID of the vendor that will charge the payment                         | no       |
+| `items`              | `array`  | Objects containing information about the items that are being purchased         | no       |
+| `fundTxData`         | `object` | Object containing information about the funding of an shopper                   | no       |
+| `genericTransactions`| `array`  | Objects containing information about every transaction that should be executed  | no       |
+| `totalAmount`        | `decimal` | The total amount of the payment, represented in the selected currency          | no       |
```
Comment (bakasura980) [inline API-Documentation.md:375]: Remove and add PaymentDetails object

File: API-Documentation.md [hunk 37]
```diff
@@ -0,0 +1,415 @@
...
+
+
+### Item
```
Comment (bakasura980) [inline API-Documentation.md:378]: Rename it to be `Payment Item`

File: API-Documentation.md [hunk 38]
```diff
@@ -0,0 +1,415 @@
...
+
+| Attribute            | Type     | Description                                                                     | Nullable |
+| -------------------- | -------- | ------------------------------------------------------------------------------- | -------- |
+| `description`        | `string` | Description of the item that is being bought. This information is displayed in the invoice. | no       |
+| `amount`             | `decimal`| Fiat amount of the item                                                         | no       |
```
Comment (bakasura980) [inline API-Documentation.md]: rename it to be `lineAmount`

Comment (bakasura980) [reply API-Documentation.md]: Also we have to tell that the amount is the total amount of the line without VAT

File: API-Documentation.md (hunk 12 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:43]: live discussion

File: API-Documentation.md (hunk 13 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:51]: live discussion

File: API-Documentation.md (hunk 14 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:59]: live discussion

File: API-Documentation.md (hunk 16 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:102]: live discussion

File: API-Documentation.md (hunk 17 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:110]: live discussion

File: API-Documentation.md (hunk 25 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:189]: live discussion

File: API-Documentation.md (hunk 24 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:198]: live discussion

File: API-Documentation.md (hunk 29 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md]: live discussion

File: API-Documentation.md (hunk 30 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md]: live discussion

File: API-Documentation.md (hunk 37 above)
Comment (Daniel-K-Ivanov) [reply API-Documentation.md:378]: live discussion

**Pull request merged**